### DISCUSSION TITLE
20230914 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#59] [Meetings]
Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x 
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Index Backend

- New feature: [Store original info-hashes pointing to the canonical one in a new table](https://github.com/torrust/torrust-index-backend/pull/269).
- New feature: for API GET endpoints (torrent details and torrent download), allow using the non-canonical info-hash, [redirecting to the canonical one](https://github.com/torrust/torrust-index-backend/pull/281).
- Fix bug: different info-hash for uploaded and downloaded torrent when the uploaded one does not have the field `private` in the `info` dict. See https://github.com/torrust/torrust-index-backend/pull/286
- Fix: [`docker compose` configuration for the Tracker](https://github.com/torrust/torrust-index-backend/pull/270). The container was updated with breaking changes.

### Torrust Index Backend

- Fix: [`docker compose` configuration for the Tracker](https://github.com/torrust/torrust-index-frontend/pull/264). The container was updated with breaking changes.

### Next

- https://github.com/torrust/torrust-index-backend/issues/290. Waiting for @da2ce7's feedback.
- Continue fixing [bugs](https://github.com/torrust/torrust-index-backend/issues?q=is%3Aissue+is%3Aopen+label%3Abug) on the Index Backend.

**Comment by mario-nt** [comment]
**Frontend**

- Closed https://github.com/torrust/torrust-index-frontend/issues/247
- Opened and worked on https://github.com/torrust/torrust-index-frontend/issues/257
- Worked on https://github.com/torrust/torrust-index-frontend/issues/205
- Opened https://github.com/torrust/torrust-index-frontend/discussions/267

**Backend**

- Opened https://github.com/torrust/torrust-index-backend/issues/273 with the help of @josecelano 
- Closed https://github.com/torrust/torrust-index-backend/issues/264

**Tracker**

- Fix [r2d2][ERROR] unable to open database file: ./storage/tracker/lib/database/sqlite3.db after syncing my froked repo with the original, to make it work again.

**Api and Types libs**

- Worked on https://github.com/torrust/torrust-index-api-lib/issues/33 with @josecelano 
- Revisit https://github.com/torrust/torrust-index-types-lib/issues/9 with @josecelano 

**Other**

- Worked on https://github.com/orgs/torrust/discussions/13#discussioncomment-6977753
- Research on the BitTorrent protocol, .torrent files. trackers, indexes.
- Discussions with @josecelano about different topics

**Comment by grmbyrn** [comment]
## SvelteKit
* Continued with Secure Git Guide.

## Torrust
* Cloned Torrust repo to inspect how an example of SvelteKit is used in a project.

## Medium
* Article on SvelteKit https://medium.com/@grmbrn89/sveltekit-project-8375efb0c293

## AI
* Course learning more about AI and how to create apps with it.
* Process for setting up an API key with Open AI.

## Other
* Organised a meeting with Jose to discuss current work.
